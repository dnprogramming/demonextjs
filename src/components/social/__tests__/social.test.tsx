import { render, screen } from "@testing-library/react";
import Social from "../social";
import React from "react";
import SocialContent from "@/content/social";

describe("Social Component", () => {
    it("renders Facebook link", () => {
        render(<Social socials={SocialContent} />);
        const facebookLink = screen.getByRole("link", {
            name: /facebook/i,
        });
        expect(facebookLink).toBeInTheDocument();
    });

    it("renders Instagram link", () => {
        render(<Social socials={SocialContent} />);
        const instagramLink = screen.getByRole("link", {
            name: /instagram/i,
        });
        expect(instagramLink).toBeInTheDocument();
    });

    it("renders social component unchanged", () => {
        const { container } = render(<Social socials={SocialContent} />);
        expect(container).toMatchSnapshot();
    });
});