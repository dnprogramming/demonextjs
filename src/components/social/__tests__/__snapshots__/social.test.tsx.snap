// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Social Component renders social component unchanged 1`] = `
<div>
  <div
    class="social"
  >
    <a
      href="https://www.facebook.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Facebook"
        data-nimg="1"
        decoding="async"
        height="48"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fsocial%2Ffb.png&w=96&q=75"
        srcset="/_next/image?url=%2Fimages%2Fsocial%2Ffb.png&w=48&q=75 1x, /_next/image?url=%2Fimages%2Fsocial%2Ffb.png&w=96&q=75 2x"
        style="color: transparent;"
        width="48"
      />
    </a>
    <a
      href="https://www.instagram.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Instagram"
        data-nimg="1"
        decoding="async"
        height="48"
        loading="lazy"
        src="/images/social/instagram.svg"
        style="color: transparent;"
        width="48"
      />
    </a>
  </div>
</div>
`;
